File: bindings/lua/Polycode/Shader.lua
require "Polycode/Resource"

class "Shader" (Resource)


function Shader:__getvar(name)
	if name == "numSpotLights" then
		return Polycode.Shader_get_numSpotLights(self.__ptr)
	elseif name == "numPointLights" then
		return Polycode.Shader_get_numPointLights(self.__ptr)
	elseif name == "screenShader" then
		return Polycode.Shader_get_screenShader(self.__ptr)
	elseif name == "name" then
		return Polycode.Shader_get_name(self.__ptr)
	end
	if Resource["__getvar"] ~= nil then
		return Resource.__getvar(self, name)
	end
end

function Shader:__setvar(name,value)
	if name == "numSpotLights" then
		Polycode.Shader_set_numSpotLights(self.__ptr, value)
		return true
	elseif name == "numPointLights" then
		Polycode.Shader_set_numPointLights(self.__ptr, value)
		return true
	elseif name == "screenShader" then
		Polycode.Shader_set_screenShader(self.__ptr, value)
		return true
	elseif name == "name" then
		Polycode.Shader_set_name(self.__ptr, value)
		return true
	end
	if Resource["__setvar"] ~= nil then
		return Resource.__setvar(self, name, value)
	else
		return false
	end
end
function Shader:getType()
	local retVal =  Polycode.Shader_getType(self.__ptr)
	return retVal
end

function Shader:setName(name)
	local retVal = Polycode.Shader_setName(self.__ptr, name)
end

function Shader:getName()
	local retVal =  Polycode.Shader_getName(self.__ptr)
	return retVal
end

function Shader:reload()
	local retVal =  Polycode.Shader_reload(self.__ptr)
end

function Shader:getExpectedParamType(name)
	local retVal = Polycode.Shader_getExpectedParamType(self.__ptr, name)
	return retVal
end

function Shader:__delete()
	if self then Polycode.delete_Shader(self.__ptr) end
end

File: bindings/javascript/Polycode/Shader.js
require('Polycode/Resource')

function Shader() {
	Object.defineProperties(this, {
		'numSpotLights': { enumerable: true, configurable: true, get: Shader.prototype.__get_numSpotLights, set: Shader.prototype.__set_numSpotLights},
		'numPointLights': { enumerable: true, configurable: true, get: Shader.prototype.__get_numPointLights, set: Shader.prototype.__set_numPointLights},
		'screenShader': { enumerable: true, configurable: true, get: Shader.prototype.__get_screenShader, set: Shader.prototype.__set_screenShader},
		'name': { enumerable: true, configurable: true, get: Shader.prototype.__get_name, set: Shader.prototype.__set_name}
	})
}


Shader.prototype = Object.create(Resource.prototype)

Shader.prototype.__get_numSpotLights = function() {
	return Polycode.Shader__get_numSpotLights(this.__ptr)
}

Shader.prototype.__set_numSpotLights = function(val) {
	Polycode.Shader__set_numSpotLights(this.__ptr, val)
}

Shader.prototype.__get_numPointLights = function() {
	return Polycode.Shader__get_numPointLights(this.__ptr)
}

Shader.prototype.__set_numPointLights = function(val) {
	Polycode.Shader__set_numPointLights(this.__ptr, val)
}

Shader.prototype.__get_screenShader = function() {
	return Polycode.Shader__get_screenShader(this.__ptr)
}

Shader.prototype.__set_screenShader = function(val) {
	Polycode.Shader__set_screenShader(this.__ptr, val)
}

Shader.prototype.__get_name = function() {
	return Polycode.Shader__get_name(this.__ptr)
}

Shader.prototype.__set_name = function(val) {
	Polycode.Shader__set_name(this.__ptr, val)
}

Duktape.fin(Shader.prototype, function (x) {
	if (x === Shader.prototype) {
		return;
	}
	Polycode.Shader__delete(x.__ptr)
})

Shader.prototype.getType = function() {
	return Polycode.Shader_getType(this.__ptr)
}

Shader.prototype.setName = function(name) {
	Polycode.Shader_setName(this.__ptr, name)
}

Shader.prototype.getName = function() {
	return Polycode.Shader_getName(this.__ptr)
}

Shader.prototype.reload = function() {
	Polycode.Shader_reload(this.__ptr)
}

Shader.prototype.getExpectedParamType = function(name) {
	return Polycode.Shader_getExpectedParamType(this.__ptr, name)
}
